activation/activation.go

```diff
@@ -244,34 +249,47 @@ func (b *Builder) buildNipstChallenge(epoch types.EpochId) error {
 func (b *Builder) StartPost(rewardAddress address.Address, dataDir string, space uint64) error {
 	b.log.Info("Starting post, reward address: %x", rewardAddress)
 	b.SetCoinbaseAccount(rewardAddress)
-	b.postInitLock.Lock()
-	if b.initStatus == Done {
-		return nil
-	}
-	if b.initStatus == InProgress {
-		b.postInitLock.Unlock()
-		return fmt.Errorf("attempted to start post when post already initiated")
+
+	initStatus := atomic.LoadInt32(&b.initStatus)
+
+	switch initStatus {
+	case InitDone:
+		return fmt.Errorf("already initialized")
```

Comment:
Why should we return an error here? why not success... we've initialized the post
also, does this cover the case where start post is called with different parameters? post will need to re run

Reply:
> Why should we return an error here? why not success... we've initialized the post

`StartPost` is async, and supposed to start something. If it's already initialized, then
nothing is going to start, and the user should know about this (at least in the api level).
Also, because it's async we're not returning successes, only validation errors. So even if it was a success, it would have created a clumsy api. In this case, the solution would be to swallow the error, to return nothing, and so to have less feedback to the user.

> also, does this cover the case where start post is called with different parameters? post will need to re run

`datadir` is being set from the user input. 
`space`- if there are no init files, the user `space` will apply. if there are, then he will get a `config mismatch` error. IMO we should not delete the files for the user implicitly. He should do so with a different api call.

Reply:
> Personally, I agree with @antonlerner. If the client sends StartMining request and gets an error, he might assume that there is no mining going on. However, I talked about it with @IlyaVi, and he pointed out that before calling StartMining the client asks for the mining status, which should indicate a running init, and therefore it is indeed an error to re-run the mining.

He wouldn’t assume that there’s no mining going if he’ll get the proper error msg.